### PR TITLE
Bind notebook editor to search model

### LINKED ISSUE #148068: Improve workspace text search with notebook editors
Made some improvements in https://github.com/microsoft/vscode/issues/95275, this issue is to continue the work

- When doing a text search with an open notebook editor, sync find matches back into the search view, same as it works for text editors
Notes

- The SearchModel needs to listen to events for notebook editors, and bind to a notebook editor, not a text model as in normal text editors, because this is where find is implemented in notebooks.
- The SearchModel needs to be fully aware of notebook cells. Match ids need to include the match cell fragment so they don't conflict between cells

### PR DESCRIPTION
**Pull request by andreamah**
Fixes #148068

### CODE REVIEW

File: src/vs/workbench/contrib/search/browser/searchModel.ts [hunk 1]
```diff
@@ -1168,11 +1214,61 @@ export class SearchResult extends Disposable {
 
 		return retEvent;
 	}
+
+	private onDidVisibleEditorsChange(): void {
+		const visibleNotebookEditors = this.editorService.visibleEditorPanes
```
Comment (rebornix) [inline src/vs/workbench/contrib/search/browser/searchModel.ts:1219]: Notebook editor widgets can be invisible and alive, I wonder if visible notebook editors are too limited.

File: src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts [hunk 2]
```diff
@@ -2384,6 +2384,7 @@ export class NotebookEditorWidget extends Disposable implements INotebookEditorD
 	}
 
 	async find(query: string, options: INotebookSearchOptions, token: CancellationToken, skipWarmup: boolean = false): Promise<CellFindMatchWithIndex[]> {
+		// here
```
[the remark this answers is not in the document]
Comment (rebornix) [reply src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts:2387]: I wonder if we need to support find without revealing the find widget.

[the remark this answers is not in the document]
Comment (andreamah) [reply src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts:2387]: @rebornix  nooo, this was supposed to be a draft! sorry about that!

**Comment by andreamah** [comment]
recreated in #167952 for better tracking locally in gh issues extension

**Pull request closed without merging**